src/matrixCells/makeMatrixArgs.js

```diff
@@ -5,24 +5,56 @@ import make_position_arr from "./makePositionArr";
 export default function make_matrix_args(regl, store) {
   // make arrays
   const arrs = {};
+  const row_nodes = store.select("network").row_nodes;
+  const col_nodes = store.select("network").col_nodes;
+  const {
+    highlighted_cols,
+    highlighted_rows,
+  } = store.select("search");
   arrs.opacity_arr = make_opacity_arr(store);
-  arrs.position_arr = {};
-  arrs.position_arr.ini = make_position_arr(
+  arrs.mask_arr = new Array(arrs.opacity_arr.length).fill(0.0);
+
+  const ini_dict = make_position_arr(
     store,
     store.select("order").inst.row,
     store.select("order").inst.col
   );
-  arrs.position_arr.new = make_position_arr(
+  const new_dict = make_position_arr(
     store,
     store.select("order").new.row,
     store.select("order").new.col
   );
-  store.dispatch(store.actions.setArrsState(arrs));
+  arrs.position_arr = {};
+  arrs.position_arr.ini = Object.values(ini_dict);
+  arrs.position_arr.new = Object.values(new_dict);
+
+  for (let i = 0; i < highlighted_cols.length; i++) {
+    if (highlighted_cols[i].length == 0) continue
+    for (let j = 0; j < highlighted_rows.length; j++) {
+      if (highlighted_rows[j].length == 0) continue
+      const targetNode = new_dict[`${row_nodes[0].name}, ${col_nodes[0].name}`];
 
+      for (let i = 0; i < arrs.mask_arr.length; i++) {
+        if (arrs.position_arr.new[i][0] === targetNode[0]) {
+          arrs.mask_arr[i] += 0.3;
+        }
+        if (arrs.position_arr.new[i][1] === targetNode[1]) {
+          arrs.mask_arr[i] += 0.3;
+        }
+        arrs.mask_arr[i] = Math.min(arrs.mask_arr[i], 1.0);
+      }
+    }
+  }
+
+  store.dispatch(store.actions.setArrsState(arrs));
   const opacity_buffer = regl.buffer({
     type: "float",
     usage: "dynamic",
   })(arrs.opacity_arr);
+  const mask_buffer = regl.buffer({
```

Comment:
What does this buffer do?

Reply:
This will pass to our shader and used for rendering the highlight zone

Reply:
Why do we need this in addition to the opacity buffer? Are we doing more for highlighting than just increasing the opacity of the selected row/col nodes or decreasing the opacity everywhere else?

Reply:
We are actaully overlay a new color instead of changing opacity now

Reply:
Ah, interesting. Do you have a screenshot?

Reply:
<img width="379" alt="image" src="https://user-images.githubusercontent.com/23213772/191564899-12246825-d3dc-4953-8346-1f59da2c8921.png">

Reply:
We should also allow users to pass a customized highlight color in this casae